src/test/java/de/tum/in/www1/artemis/programmingexercise/ProgrammingSubmissionIntegrationTest.java

```diff
@@ -386,7 +386,7 @@ public void getProgrammingSubmissionWithoutAssessment_alreadyAssessed_noFound()
         final var tutor = database.getUserByLogin("tutor1");
         database.addResultToSubmission(submission, AssessmentType.SEMI_AUTOMATIC, tutor);
 
-        request.get("/api/exercises" + exercise.getId() + "/programming-submission-without-assessment", HttpStatus.NOT_FOUND, String.class);
```

Comment:
This did only return 404 before, because `/api/exercises1234/..` is not valid

Reply:
This is heavily broken. 985e31f7 updated the test case to incorporate the due date check implemented by @krusche in July.